### PR TITLE
[docs] Improve FAQ

### PR DESCRIPTION
**Pull request by mattstein**
## The Problem/Issue/Bug:

The [current FAQ](https://ddev.readthedocs.io/en/stable/users/basics/faq/) is a bit sprawling, and it contains important information that can’t be linked to directly.

## How this PR Solves The Problem:

This shapes up the FAQ to be more readable and linkable in a few ways:

- Re-titles excessively-explanatory “Frequently-Asked Questions (FAQ)” simply to “FAQ”, while repurposing the phrase “frequently-asked questions” in introductory framing for the one person who searches that term or has never seen an FAQ.
- Gives a proper heading to each Q&A item, further organized categorically for easier browsing and more orderly maintenance.
- Pares down headings wherever possible for brevity and readability, with sense breaks added to text.
- Swaps in the word “workstation” in a few places, since we’re trying to use that more consistently.
- Keeps all existing Q&A items, paring down some in favor of linking (which reinforces documentation structure), and adding some explanatory or illustrative content where more context can be useful. (Most obviously in _How can I connect to my database?_, which I assume is a constant point of friction for any container-based development environment.)
- Rewords the question about contributing financially so it’s less redundant with the preceding question. (This content should get a landing page with more detail on ddev.com!)

This comes with the downside of chaotic sidebar navigation, which I’ve addressed separately in #4294.

## Manual Testing Instructions:

https://ddev.readthedocs.io/en/stable/developers/testing-docs/#preview-changes

## Automated Testing Overview:

n/a

## Related Issue Link(s):

Part of this overall effort:

- #4188

## Release/Deployment notes:

n/a

<a href="https://gitpod.io/#https://github.com/drud/ddev/pull/4293"></a>

### CODE REVIEW

**Comment by rfay** [comment]
Thanks, probably won't be able to review until next week, but much appreciated!

File: docs/content/users/basics/faq.md [hunk 1]
```diff
@@ -1,144 +1,217 @@
-# Frequently-Asked Questions (FAQ)
+# FAQ
```
Comment (rfay) [inline docs/content/users/basics/faq.md:1]: I suspect the old way catches more SEO and provides more than one way to find this in search.

**Review by rfay — APPROVED** [review]
This is a great improvement. Thanks!

File: docs/content/users/basics/faq.md (hunk 1 above)
Comment (mattstein) [reply docs/content/users/basics/faq.md:1]: I mentioned search in the first bullet point of the PR description: I kept the phrase “frequently-asked questions” on the page with that in mind. Giving each question a proper heading is *probably* better for overall SEO surfacing those questions—and I’m guessing `ddev faq` would be a more common search than `ddev frequently asked questions`, but that could be verified with research.

The real win would be providing [structured data to further expose Q&A items](https://developers.google.com/search/docs/appearance/structured-data/faqpage), but that’d require a far deeper journey into the theme.

**Comment by rfay** [comment]
Will pull when you're ready.

**Comment by mattstein** [comment]
Pull away!

**Pull request merged**